CorsixTH/Lua/app.lua

```diff
@@ -399,23 +400,38 @@ end
 
 --! Tries to initialize the user and built in level directories, returns true on
 --! success and false on failure.
-function App:initUserLevelDir()
+-- TODO: Integrate other directory initialisations into this function
+function App:initUserDirectories()
   local conf_path = self.command_line["config-file"] or "config.txt"
-  self.user_level_dir = self.config.levels or
-      conf_path:match("^(.-)[^" .. pathsep .. "]*$") .. "Levels"
 
-  if self.user_level_dir:sub(-1, -1) == pathsep then
-    self.user_level_dir = self.user_level_dir:sub(1, -2)
-  end
-  if lfs.attributes(self.user_level_dir, "mode") ~= "directory" then
-    if not lfs.mkdir(self.user_level_dir) then
-      print("Notice: Level directory does not exist and could not be created.")
-      return false
+  --! Attempt to set the user's directory choice
+  --!param dir (path) The defined path of the folder by the user
+  --!param label (string) What folder is being set, for the error dialogue
+  --!return The fully qualified path or 'false' if directory creation failed (leading
+  --! to game exit)
+  local function setUserDir(dir, label)
+    if dir:sub(-1, -1) == pathsep then
+      dir = dir:sub(1, -2)
     end
+    if lfs.attributes(dir, "mode") ~= "directory" then
+      if not lfs.mkdir(dir) then
+        print("Notice:" .. label .. "directory does not exist and could not be created.")
+        return false -- Force game exit
```

Comment:
This use to return false from the outer function, now the field gets set to false instead.

The new comment about forcing the game to exit doesn't appear to be credible, or at least direct.